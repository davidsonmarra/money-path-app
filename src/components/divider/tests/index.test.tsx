import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Divider, {
  DividerType,
  DividerSize,
  Props as DividerProps,
} from 'src/components/divider';

const renderComponent = ({
  type = DividerType.horizontal,
  size = DividerSize.medium,
  ...rest
}: Partial<DividerProps>) =>
  render(<Divider type={type} size={size} {...rest} />);

describe('Divider', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Horizontal', () => {
    it('renders horizontal divider', () => {
      const divider = renderComponent({type: DividerType.horizontal}).root;
      expect(divider.props.style.width).toBe('100%');
    });

    it('renders small horizontal divider', () => {
      const divider = renderComponent({size: DividerSize.small}).root;
      expect(divider.props.style.height).toBe(1);
    });

    it('renders medium horizontal divider', () => {
      const divider = renderComponent({size: DividerSize.medium}).root;
      expect(divider.props.style.height).toBe(2.5);
    });

    it('renders large horizontal divider', () => {
      const divider = renderComponent({size: DividerSize.large}).root;
      expect(divider.props.style.height).toBe(5);
    });
  });

  describe('Vertical', () => {
    it('renders vertical divider', () => {
      const divider = renderComponent({type: DividerType.vertical}).root;
      expect(divider.props.style.height).toBe('100%');
    });

    it('renders small vertical divider', () => {
      const divider = renderComponent({
        type: DividerType.vertical,
        size: DividerSize.small,
      }).root;
      expect(divider.props.style.width).toBe(1);
    });

    it('renders medium vertical divider', () => {
      const divider = renderComponent({
        type: DividerType.vertical,
        size: DividerSize.medium,
      }).root;
      expect(divider.props.style.width).toBe(2.5);
    });

    it('renders large vertical divider', () => {
      const divider = renderComponent({
        type: DividerType.vertical,
        size: DividerSize.large,
      }).root;
      expect(divider.props.style.width).toBe(5);
    });
  });
});
