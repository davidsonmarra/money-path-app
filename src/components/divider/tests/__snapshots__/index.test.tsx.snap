// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF33",
      "height": 2,
      "width": "100%",
    }
  }
/>
`;
